spec/services/analytics_spec.rb

```diff
@@ -207,22 +173,9 @@
         ahoy: ahoy,
       )
 
-      analytics_hash = {
-        event_properties: {},
-        user_id: current_user.uuid,
-        locale: I18n.locale,
-        git_sha: IdentityConfig::GIT_SHA,
-        git_branch: IdentityConfig::GIT_BRANCH,
-        new_session_success_state: true,
-        success_state: success_state,
-        new_session_path: true,
-        new_event: true,
-        path: path,
-        session_duration: 7.0,
-      }
-
-      expect(ahoy).to receive(:track).
-        with('Trackable Event', analytics_hash.merge(request_attributes))
+      analytics_attributes[:session_duration] = 7.0
+
+      expect(ahoy).to receive(:track).with('Trackable Event', analytics_attributes)
```

Comment:
I think we could avoid mutation here as well
```suggestion
      expect(ahoy).to receive(:track).with(
        'Trackable Event',
        analytics_attributes.merge(session_duration: 7.0),
      )
```

Reply:
I think we have a different philosophy of clarifying in a test example, what is specifically different about the example (compared with other examples). I can see your point about mutations; seems like mutation (of let variable here) is acceptable in test examples.  I will remove all of the (other) mutations, and pass the updated hash to the expectation (as you are recommending).

On another note, the GitLab pipeline has failed twice: first on a test file (add_email_spec.rb) that is unrelated to this PR, now on Docker not able to pull Postgres 13.4.  Am I expected to continue with dummy commits, until the pipeline passes? Let me know if I am missing something about the CI pipeline tests. Thank you!

Reply:
yes unfortunately if the specs keep flaking in CI, clicking retry or adding empty commits is the best day, I'd ask in #login-devops for help as well.